src/app/fdctl/run/run.c

```diff
@@ -60,145 +70,56 @@ install_tile_signals( void ) {
-  fd_log_private_group_id_set( (ulong)pid );
-  FD_LOG_NOTICE(( "booting tile %s(%lu) pid(%d)", fd_topo_tile_kind_str( tile->kind ), tile->kind_id, pid ));
-
-  install_tile_signals();
-
-  /* calling fd_tempo_tick_per_ns requires nanosleep, it is cached with
-     a FD_ONCE.  We do this for all tiles before sandboxing so that we
-     don't need to allow the nanosleep syscall. */
-  fd_tempo_tick_per_ns( NULL );
-
-  /* preload shared memory before sandboxing, so it is already mapped */
-  fd_topo_join_tile_workspaces( args->config->name, &args->config->topo, tile );
-
-  fd_tile_config_t * config = fd_topo_tile_to_config( tile );
-
-  void * scratch_mem   = NULL;
-  if( FD_LIKELY( config->scratch_align ) ) {
-    scratch_mem = (uchar*)args->config->topo.workspaces[ tile->wksp_id ].wksp + tile->user_mem_offset;
-    if( FD_UNLIKELY( !fd_ulong_is_aligned( (ulong)scratch_mem, config->scratch_align() ) ) )
-      FD_LOG_ERR(( "scratch_mem is not aligned to %lu", config->scratch_align() ));
-  }
-
-  if( FD_UNLIKELY( config->privileged_init ) )
-    config->privileged_init( &args->config->topo, tile, scratch_mem );
+extern int * fd_log_private_shared_lock;
 
-  int allow_fds[ 32 ];
-  ulong allow_fds_cnt = config->populate_allowed_fds( scratch_mem,
-                                                      sizeof(allow_fds)/sizeof(allow_fds[ 0 ]),
-                                                      allow_fds );
+static int write_config_memfd( config_t * config ) {
+  int config_memfd = memfd_create( "config", 0 );
+  if( FD_UNLIKELY( -1==config_memfd ) ) FD_LOG_ERR(( "memfd_create() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  if( FD_UNLIKELY( -1==ftruncate( config_memfd, sizeof( config_t ) ) ) ) FD_LOG_ERR(( "ftruncate() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
 
-  struct sock_filter seccomp_filter[ 128UL ];
-  ulong seccomp_filter_cnt = config->populate_allowed_seccomp( scratch_mem,
-                                                               sizeof(seccomp_filter)/sizeof(seccomp_filter[ 0 ]),
-                                                               seccomp_filter );
-  fd_sandbox( args->config->development.sandbox,
-              args->config->uid,
-              args->config->gid,
-              allow_fds_cnt,
-              allow_fds,
-              seccomp_filter_cnt,
-              seccomp_filter );
-
-  /* Now we are sandboxed, join all the tango IPC objects in the workspaces */
-  fd_topo_fill_tile( &args->config->topo, tile, FD_TOPO_FILL_MODE_JOIN );
-  FD_TEST( tile->cnc );
+  uchar * bytes = mmap( NULL, sizeof( config_t ), PROT_READ | PROT_WRITE, MAP_SHARED, config_memfd, 0 );
+  if( FD_UNLIKELY( bytes == MAP_FAILED ) ) FD_LOG_ERR(( "mmap() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  fd_memcpy( bytes, config, sizeof( config_t ) );
+  if( FD_UNLIKELY( munmap( bytes, sizeof( config_t ) ) ) ) FD_LOG_ERR(( "munmap() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
```

Comment:
It's probably not necessary since the tiles close the memfd early on in the boot process and map it readonly, but you could seal the memfd at this point to prevent writing for paranoia

Reply:
Not needed imo